src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -105,15 +112,9 @@ public KafkaTopicRepository(final ZooKeeperHolder zkFactory, final KafkaFactory
     public void createTopic(final EventType eventType) throws TopicCreationException, DuplicatedEventTypeNameException {
         createTopic(eventType.getTopic(),
                 calculateKafkaPartitionCount(eventType.getDefaultStatistic()),
-                settings.getDefaultTopicReplicaFactor(),
-                getTopicRetentionMs(eventType),
-                settings.getDefaultTopicRotationMs());
-    }
-
-    private long getTopicRetentionMs(final EventType eventType) {
-        return Optional.ofNullable(eventType.getOptions())
-                .map(EventTypeOptions::getRetentionTime)
-                .orElse(settings.getDefaultTopicRetentionMs());
+                nakadiSettings.getDefaultTopicReplicaFactor(),
+                eventType.getOptions().getRetentionTime(),
```

Comment:
It should not be the responsibility of caller. He sets null to nullable property, he does everything correct. He may not be aware that createTopic requires it to be not null.
Also, the null will not come to kafka, there will be NullPointerException right on this line when it will try to unbox Long to long.